node/src/test_utils/recorder.rs

```diff
@@ -271,6 +302,16 @@ impl Recording {
             .unwrap_or_else(|e| panic!("{}", e))
     }
 
+    pub fn get_record_partial_eq_less<T>(&self, index: usize) -> &T
+    where
+        T: Any + Send,
+    {
+        let wrapped_msg: &PretendedMatchable<T> = self
+            .get_record_inner_body(index)
+            .unwrap_or_else(|e| panic!("{}", e));
+        &wrapped_msg.0
+    }
+
     pub fn get_record_opt<T>(&self, index: usize) -> Option<&T>
     where
         T: Any + Send,
```

Comment:
I can't mark as low as I'd like, but we talked about modifying the downcasting part of `get_record_inner_body` to catch downcasting errors and check to see: if it doesn't downcast to `T`, does it downcast to `PretendedMatchable<T>`?

Then you wouldn't need a special version of `get_record()`.